bonus_guides/G_static_assets.md

```diff
@@ -1 +1,289 @@
+
+Take notice of `"web/static/js/app"`. This is not really a file name, this is the name of a module into which Brunch wrapped the code in `"web/static/js/app.js"`
+
+
+Let's add one more file `web/static/js/greeter.js`:
+
+```javascript
+export var Greet = {
+  greet: function(){
+    console.log("Hello!")
+  }
+}
+
+export var Bye = {
+  greet: function(){
+    console.log("Bye!")
+  }
+}
+```
+
+and modify `web/static/js/app.js` to require the new module:
+
+```javascript
+import { Greet } from "web/static/js/greeter";
+
+export var App = {
+  run: function(){
+    Greet.greet()
+  }
+}
+```
+
+Please reload the page. We should see `Hello!` in the browser Javascript console.
+
+Object `Bye` was not imported into package `"web/static/js/app"`, even though `Bye` is declared as exportable.
+
+Please pay attention to how differently we required module `web/static/js/app.js` from the HTML page, and how we imported module `web/static/js/greeter` from `web/static/js/app`. This is because there is no preprocessing happening for HTML pages and we cannot use ES6 syntax.
+
+
+#### Legacy Non-modularized Code
+
+If we have some legacy Javascript code which doesn't play well with module systems and/or we need global variables it defines, all we need to do is place our Javascript into directory `web/static/vendor`. Brunch will not wrap these files in modules.
+
+Let's test it. Create `web/static/vendor` if it does not exist yet and create file `web/static/vendor/meaning_of_life.js` with just one line in it:
+
+```
+meaning_of_life = 42;
+```
+
+Reload the page. Open the JS console and type `meaning_of_life`. This will return `42`. The variable is global.
+
+Important detail: according to the default configuration there is no ES6 support for files in `web/static/vendor`. Should we need to enable it, look for `plugins: { babel: { ignore:` in `brunch-config.js`.
+
+#### Brunch Plugin Pipeline
+
+All transformations Brunch performs are actually done by plugins. Brunch automatically uses installed plugins listed in  `package.json`. Here is what the pipeline looks like for a  newly generated Phoenix project:
+
+##### Javascript
+
+- [`babel-brunch`](https://github.com/babel/babel-brunch) transpiles ES6 code to vanilla ES5 Javascript using [Babel](https://github.com/babel/babel)
+- [`javascript-brunch`](https://github.com/brunch/javascript-brunch) is the main Javascript processing plugin. Without it Javascript files will be ignored.
+- [`uglify-js-brunch`](https://github.com/brunch/uglify-js-brunch) minifies Javascript code
+
+##### CSS
+
+- [`clean-css-brunch`](https://github.com/brunch/clean-css-brunch) is a minifier for CSS
+- [`css-brunch`](https://github.com/brunch/css-brunch) is the main CSS processing plugin. Without it  CSS files will be ignored.
+
+
+It is very easy to add more plugins. We can find a plethora of Brunch plugins [here](http://brunch.io/plugins.html) and [here](https://www.npmjs.com/search?q=brunch).
```

Comment:
Rather than the links being "here" and "here", could they be named? Maybe "find a plethora of Brunc plugins on the [brunch website] and on [npm]"?